core/src/main/java/com/linecorp/armeria/common/util/CompositeException.java

```diff
@@ -155,12 +163,18 @@ public synchronized Throwable getCause() { // NOPMD
                             aggregateMessage.append(separator);
                         }
 
-                        for (int i = 0; i < depth + 2; i++) {
-                            aggregateMessage.append("  ");
-                        }
                         final StackTraceElement[] st = inner.getStackTrace();
                         if (st.length > 0) {
-                            aggregateMessage.append("at ").append(st[0]).append(separator);
+                            final boolean isEnableVerboseExceptionFlag =
+                                    verboseExceptionFlag.isSampled(inner.getClass());
+                            final int loopCount = isEnableVerboseExceptionFlag ?
+                                  Math.min(st.length, maxStacktraceSize) : disabledVerboseExceptionLoopCount;
```

Comment:
I don't see that this correctly applies the contract in #4347
- If `verboseExceptionFlag.isSampled(inner.getClass())` is `true`, 
  full stack traces should be appended.
- If `verboseExceptionFlag.isSampled(inner.getClass())` is `false`, 
  at most 20 stack frames could be appended.

Reply:
Should we add a test for the change?

Reply:
> I don't see that this correctly applies the contract in #4347
> 
> * If `verboseExceptionFlag.isSampled(inner.getClass())` is `true`,
>   full stack traces should be appended.
> * If `verboseExceptionFlag.isSampled(inner.getClass())` is `false`,
>   at most 20 stack frames could be appended.

Sorry I misunderstood.  this will be fixed soon

Reply:
> Should we add a test for the change?

I think testing is necessary.  Here's the scenario I was thinking of:  Set the value of `verboseExceptions` using `System.setProperty` in unit tests.  As a result, the `ExceptionOverview` created by calling `getCause()` of the `CompositeException` class is obtained, the stack trace String is cut with `line.separator`, and the output stack trace length is compared.  What do you think about it?

Reply:
Since the stack trace is output differently for each `Flag`(**verboseExceptions**), I think a test to compare the number of output lines is necessary.

Reply:
> . As a result, the ExceptionOverview created by calling getCause() of the CompositeException class is obtained, the stack trace String is cut with line.separator, and the output stack trace length is compared. What do you think about it?

Sounds good.

For the convenience of tests, we can add a package-private constructor like:
```java
    public CompositeException(Iterable<? extends Throwable> errors) {
        this(errors, Flags.verboseExceptionSampler());
    }

    @VisibleForTesting
    CompositeException(Iterable<? extends Throwable> errors, Sampler<Class<? extends Throwable>> verboseExceptionSampler) {
        ...
    }
```